### PR TITLE
Replace master with specific jaeger tag version in otel-collector

### LINKED ISSUE #2617: Otel-collector example refers to non-existing raw files
The [Makefile](https://github.com/open-telemetry/opentelemetry-go/blob/main/example/otel-collector/Makefile) in the otel-collector example uses deleted yamls for `jaeger-operator-k8s`:
```
jaeger-operator-k8s:
	# Create the jaeger operator and necessary artifacts in ns observability
	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/crds/jaegertracing.io_jaegers_crd.yaml
	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/service_account.yaml
	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/role.yaml
	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/role_binding.yaml
	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/operator.yaml
```

None of those raw links is valid anymore.

### PR DESCRIPTION
**Pull request by hanyuancheung**
Fix #2617 
- Cannot use master as the "version", instead of using pined versions.
- Replace master with specific jaeger tag version in otel-collector example.

### CODE REVIEW

File: example/otel-collector/Makefile [hunk 1]
```diff
@@ -3,15 +3,15 @@ namespace-k8s:
 
 jaeger-operator-k8s:
 	# Create the jaeger operator and necessary artifacts in ns observability
-	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/crds/jaegertracing.io_jaegers_crd.yaml
-	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/service_account.yaml
-	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/role.yaml
-	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/role_binding.yaml
-	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/operator.yaml
+	kubectl create -n observability -f https://raw.githubusercontent.com/jaegertracing/jaeger-operator/v1.31.0/deploy/crds/jaegertracing.io_jaegers_crd.yaml
```
Comment (pellared) [inline example/otel-collector/Makefile:6]: Try https://raw.githubusercontent.com/jaegertracing/jaeger-operator/v1.31.0/deploy/crds/jaegertracing.io_jaegers_crd.yaml 😄 

Check https://github.com/jaegertracing/documentation/pull/542/files

**Review by pellared — CHANGES_REQUESTED** [review]
The URLs are invalid

File: example/otel-collector/Makefile (hunk 1 above)
Comment (hanyuancheung) [reply example/otel-collector/Makefile]: sorry about that, I've updated the version info.

Comment (pellared) [reply example/otel-collector/Makefile]: Any reason why not using https://github.com/jaegertracing/jaeger-operator/releases/download/v1.31.0/jaeger-operator.yaml as documented?

**Comment by dmathieu** [comment]
This seems a bit brittle, and we'll have to make sure we don't go too much behind on the version used.
How about rewording the example so that it points directly to jaeger's documentation instead of duplicating it here?

**Review by Aneurysm9 — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Comment by MrAlias** [comment]
@hanyuancheung can you address these unresolved issues so we can move to merging this?

- https://github.com/open-telemetry/opentelemetry-go/pull/2622#discussion_r809083136
- https://github.com/open-telemetry/opentelemetry-go/pull/2622#issuecomment-1042993746

**Comment by hanyuancheung** [comment]
> @hanyuancheung can you address these unresolved issues so we can move to merging this?
> 
> * [Replace master with specific jaeger tag version in otel-collector #2622 (comment)](https://github.com/open-telemetry/opentelemetry-go/pull/2622#discussion_r809083136)
> * [Replace master with specific jaeger tag version in otel-collector #2622 (comment)](https://github.com/open-telemetry/opentelemetry-go/pull/2622#issuecomment-1042993746)

Sure. How about updating the jaeger operator version to v1.31.0, and pointing to the jaeger's documentation via an annotation in makefile?

File: example/otel-collector/Makefile (hunk 1 above)
Comment (MrAlias) [reply example/otel-collector/Makefile]: @pellared please take another look.

**Comment by MrAlias** [comment]
@dmathieu please take another look.

**Review by pellared — APPROVED** [review]
Looks OK (but I have not tested it 😬 ).

**Review by dmathieu — COMMENTED** [review]
I haven't tested it either. But it looks good to me.
